semantic-core/src/Data/Term.hs

```diff
@@ -48,46 +41,3 @@ instance RightModule sig => Monad (Term sig) where
 
 instance RightModule sig => Carrier sig (Term sig) where
   eff = Term
-
-
-class (HFunctor sig, forall g . Functor g => Functor (sig g)) => Syntax sig where
-  foldSyntax :: (forall x y . (x -> m y) -> f x -> n y)
-             -> (forall a . Incr () (n a) -> m (Incr () (n a)))
-             -> (a -> m b)
-             -> sig f a
-             -> sig n b
-
-instance Syntax (Scope ()) where
-  foldSyntax go bound free = Scope . go (bound . fmap (go free)) . unScope
-
-instance (Syntax l, Syntax r) => Syntax (l :+: r) where
-  foldSyntax go bound free (L l) = L (foldSyntax go bound free l)
-  foldSyntax go bound free (R r) = R (foldSyntax go bound free r)
-
-
-iter :: forall m n sig a b
-     .  Syntax sig
-     => (forall a . m a -> n a)
-     -> (forall a . sig n a -> n a)
-     -> (forall a . Incr () (n a) -> m (Incr () (n a)))
-     -> (a -> m b)
-     -> Term sig a
-     -> n b
-iter var alg bound = go
-  where go :: forall x y . (x -> m y) -> Term sig x -> n y
-        go free = \case
-          Var a -> var (free a)
-          Term t -> alg (foldSyntax go bound free t)
-
-cata :: Syntax sig
-     => (a -> b)
-     -> (forall a . sig (Const b) a -> b)
-     -> (Incr () b -> a)
-     -> (x -> a)
-     -> Term sig x
-     -> b
-cata var alg k h = getConst . iter (coerce var) (Const . alg) (coerce k) (Const . h)
-
-
-interpret :: (Carrier sig m, Member eff sig, Syntax eff) => (forall a . Incr () (m a) -> m (Incr () (m a))) -> (a -> m b) -> Term eff a -> m b
-interpret = iter id send
```

Comment:
If we want to implement something like `interpret` again in the future, it should be easy enough since it shouldn’t actually require a `Syntax` instance to walk the `Term` direct-recursively, just an `HFunctor` instance.